main.py

```diff
@@ -26,59 +31,79 @@ def validate_number(num):
         return False
 
 
-def validate_operator(operator):
+def validate_operator(operator: any) -> bool:
+    """Validate value for being in a list of operators and print a massage if the validation failed."""
     if operator in valid_operators:
         return True
 
     print(f'Please enter one on these operators: {", ".join(valid_operators)}')
     return False
 
 
-def handle_input(phrase, validator):
+def handle_input(phrase: str, validator: Callable[[any], bool]) -> T:
+    """
+    Wait for user's input, exit app or validate and return input
+
+    :param phrase: guiding message for the user
+    :param validator: function to validate user input
+    :return: valid user input
+    """
     res = input(phrase)
 
     if res == 'exit':
         app_exit()
-        return
 
     if not validator(res):
         res = handle_input(phrase, validator)
 
     return res
 
 
-def inputs():
+def read_user_inputs() -> None:
+    """Ask user to enter 2 numbers and operators, process them and show result, rinse and repeat"""
     a = handle_input("Enter first number:", validate_number)
     operator = handle_input("Enter the operator you want to use:", validate_operator)
     b = handle_input("Enter second number:", validate_number)
     count(a, b, operator)
+    read_user_inputs()
+
 
+def count(a: T, b: T, operator: str) -> None:
+    """
+    Perform math operation based on user inputs and print result
 
-def count(a, b, operator):
+    :param a: first number
+    :param b: second number
+    :param operator: math operator
+    """
     first_num = float(a)
     second_num = float(b)
 
     if operator == '+':
-        print("The sum is :", round(first_num + second_num, 3))
+        print("The sum is :", do_math(round(first_num + second_num, 3)))
     elif operator == '-':
-        print("The result of negation is: ", round(first_num - second_num, 3))
+        print("The result of negation is: ", do_math(round(first_num - second_num, 3)))
     elif operator == '/':
         if second_num != float(0):
-            print("The result of division: ", round(first_num / second_num, 3))
+            print("The result of division: ", do_math(round(first_num / second_num, 3)))
         else:
             print("Hold you horses, you can't divide by 0! Try again!")
-            inputs()
+            read_user_inputs()
     elif operator == '*':
-        print("The result of multiplying: ", round(first_num * second_num, 3))
+        print("The result of multiplying: ", do_math(round(first_num * second_num, 3)))
+
 
+def do_math(num: T) -> T:
+    """Check if the number is integer and convert it"""
+    return int(num) if num%1 == 0 else num
```

Comment:
This method actually doesn't do anything right now? no matter what number you take, this number % 1 will be this number (dividing by one)

Reply:
@marcinoleksiak actually, I tried to use here modulo to check if the result of the math operation is int or float, to not display .00 in the final result if they are not needed
got the inspiration here - https://stackoverflow.com/questions/4541155/check-if-a-number-is-int-or-float/44349980#44349980

Reply:
actually it's a good one. I didn't notice the trick :)